src/plugins/intel_cpu/src/memory_control.hpp

```diff
@@ -30,9 +23,16 @@ class MemoryControl {
     using MemoryBlockMap = std::unordered_map<decltype(MemoryRegion::id), MemoryBlockPtr>;
 
 public:
-    static edgeClusters findEdgeClusters(const std::vector<EdgePtr>& graphEdges);
+    static EdgeClusters formEdgeClusters(const std::vector<EdgePtr>& graphEdges);
+    static MemoryRegions formMemoryRegions(const EdgeClusters& clusters, size_t remaining, const GlobalExecutionIndex& globalExecIndex);
+    static OutputMemoryBlocks filterOutDynamicOutputEdges(MemoryRegions& memoryRegions,
+                                                                const EdgeClusters& clusters,
+                                                                const std::map<std::size_t, NodePtr>& outputNodes);
```

Comment:
Isn't it better to put these utility methods to some place other than the `MemoryControl ` class to keep the latter plugin independent?

Reply:
Agree.
I am going to move it back to the graph